sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -3410,6 +3410,13 @@ object functions {
   /**
    * Returns an array of elements after applying a transformation to each element
    * in the input array.
+   * {{{
+   *   df.select(transform(col("i"), x => x + 1))
+   *   df.selectExpr("transform(i, x -> x + 1")
```

Comment:
Actually, do we need this? `"transform(i, x -> x + 1)"` is SQL specific examples that have to be located with `ExpressionInfo`.